integration-tests/validation/src/main/java/com/android/designcompose/testapp/validation/examples/InheritanceTest.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.designcompose.testapp.validation.examples
+
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.width
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import com.android.designcompose.ComponentReplacementContext
+import com.android.designcompose.annotation.Design
+import com.android.designcompose.annotation.DesignComponent
+import com.android.designcompose.annotation.DesignDoc
+
```

Comment:
I honestly have no idea what half of the code is doing here, what is `ParentNoDesignDoc` and what is it demonstrating? I think I see what `replaceIndicator` is doing, but the naming is odd (it'd be more clear if it was `indicatorToBeOverridden` or something). And I'd really like to see a test that shows the base interface working as well.

Reply:
This is just showing that the interface `Child` can inherit from multiple interfaces. It inherits from both `ParentGen`, an interface that is generated, and `ParentNoDesignDoc`, an interface that is not generated. The generated `ChildDoc` object can then call functions from `Child`, `ParentGen`, and `ParentNoDesignDoc`. I can add what I've just written in a comment.